### ISSUE TITLE
[java] New Rule: UseTryWithResources - Replace close and IOUtils.closeQuietly with try-with-resources

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#1405]
As mentioned [here](https://github.com/pmd/pmd/pull/1401#issuecomment-431495555) on #1401, we could create a new rule, that finds usages of IOUtils.closeQuietly within a finally block.

The rule would have a minimum language version of 7.

A simple XPath expression could be:

```
//TryStatement[FinallyStatement//Name[@Image='IOUtils.closeQuietly']]
```

**Comment by Thunderforge** [comment]
`IOUtils.closeQuietly(foo)` is [from Apache Commons-IO](https://commons.apache.org/proper/commons-io/javadocs/api-release/org/apache/commons/io/IOUtils.html#closeQuietly-java.io.Closeable-). Do we typically make rules for third-party dependencies? For what it's worth, the method is deprecated as of version 2.6.

**Comment by hooperbloob** [comment]
Revising the rule to support multiple triggers would be an improvement and allow it to live beyond deprecated methods. Making it parametric and letting the user supply any additional @Image candidates that would trigger it.  UseTryWithResources might be a better, broader name.

**Comment by adangel** [comment]
That's right, we can make it more general: If a AutoClosable is explicitly closed in a finally block, the rule should be triggered. We might need to add a couple of trigger candidates ("close", "closeQuietly").
I like the name **UseTryWithResources**.

**Comment by jsotuyod** [comment]
FYI: fb-contrib has an equivalent check for Findbugs / Spotbugs: http://fb-contrib.sourceforge.net/bugdescriptions.html#UTWR_USE_TRY_WITH_RESOURCES
